ocs_ci/ocs/ui/block_pool.py

```diff
@@ -173,3 +176,130 @@ def reach_pool_limit(self, replica, compression):
+        compression_saving_value, compression_saving_unit = compression_saving.split(
+            " "
+        )
+        logger.info(
+            f"compression_saving_value {compression_saving_value} "
+            f"compression_saving_unit {compression_saving_unit}"
+        )
+        logger.info(
+            f"compression_eligibility {compression_eligibility} "
+            f"compression_ratio {compression_ratio} "
+            f" compression_saving {compression_saving}"
+        )
+
+        # Get pool id and query Prometheus
+        pool_id = get_pool_id(pool_name)
+        if pool_id is None:
+            raise PoolIdNotFound(f"Pool {pool_name} didn't return pool id")
+        new_prom = PrometheusAPI()
+        ceph_pool_compress_bytes_used_data = new_prom.query(
+            f'ceph_pool_compress_bytes_used{{pool_id="{pool_id}"}}'
+        )
+        ceph_pool_compress_under_bytes_data = new_prom.query(
+            f'ceph_pool_compress_under_bytes{{pool_id="{pool_id}"}}'
+        )
+
+        # Calculate compression savings
+        ceph_pool_compress_bytes_used_value = ceph_pool_compress_bytes_used_data[0].get(
+            "value"
+        )
+        ceph_pool_compress_under_bytes_value = ceph_pool_compress_under_bytes_data[
+            0
+        ].get("value")
+        final_saving = int(ceph_pool_compress_under_bytes_value[1]) - int(
+            ceph_pool_compress_bytes_used_value[1]
+        )
+        logger.info(f"final saving in bytes {final_saving}")
+
+        # Convert final saving to right measure
+        prometheus_final_saving_human = convert_bytes_into_right_measure(final_saving)
+        logger.info(f"Final saving in ui measure {prometheus_final_saving_human}")
+
+        # Calculate compression ratio
+        final_ratio = int(ceph_pool_compress_under_bytes_value[1]) / int(
+            ceph_pool_compress_bytes_used_value[1]
+        )
+        logger.info(f"final ratio {final_ratio}")
+
+        # Compare all parameters
+        (
+            expected_compression_value,
+            expected_compression_unit,
+        ) = expected_compression_saving.split(" ")
+
+        return_value_are_equal = True
+        prom_saving_value, prom_final_unit = prometheus_final_saving_human.split(" ")
+        if (
+            format(float(compression_saving_value), ".2f")
+            != format(float(expected_compression_value), ".2f")
+            or compression_saving_unit != expected_compression_unit
+        ):
+            return_value_are_equal = False
+            logger.info(
+                f"UI compression saving are not equal to expected saving: "
+                f"UI compression value is {compression_saving_value} "
+                f"and expected compression value is {expected_compression_value}. "
+                f"UI compression unit is {compression_saving_unit} "
+                f"and prometheus compression unit is {expected_compression_unit}"
+            )
+
+        if (
```

Comment:
shouldn't it be `elif`? do we want to have all these conditions checked? if the first sets the `return_value_are_equal` to False, isn't it enough?

Reply:
I thought to get a wider picture on all value compare before returning False. It doesn't take time as data is not retrieved but just compared.